plugins/gimp/file-jxl-load.cc

```diff
@@ -10,163 +10,252 @@
 #undef MIN
 #undef CLAMP
 
-#include "lib/jxl/alpha.h"
+#include <vector>
+
+#include "jxl/decode.h"
+#include "jxl/decode_cxx.h"
+#include "jxl/resizable_parallel_runner.h"
+#include "jxl/resizable_parallel_runner_cxx.h"
 #include "lib/jxl/base/file_io.h"
-#include "lib/jxl/base/thread_pool_internal.h"
-#include "lib/jxl/dec_file.h"
-#include "plugins/gimp/common.h"
 
 namespace jxl {
 
-namespace {
-
-template <GimpPrecision precision, bool has_alpha, size_t num_channels>
-void FillBuffer(
-    const CodecInOut& io,
-    std::vector<typename BufferFormat<precision>::Sample>* const pixel_data) {
-  pixel_data->reserve(io.xsize() * io.ysize() * (num_channels + has_alpha));
-  for (size_t y = 0; y < io.ysize(); ++y) {
-    const float* rows[num_channels];
-    for (size_t c = 0; c < num_channels; ++c) {
-      rows[c] = io.Main().color().ConstPlaneRow(c, y);
-    }
-    const float* const alpha_row =
-        has_alpha ? io.Main().alpha().ConstRow(y) : nullptr;
-    for (size_t x = 0; x < io.xsize(); ++x) {
-      const float alpha = has_alpha ? alpha_row[x] : 1.f;
-      const float alpha_multiplier =
-          has_alpha && io.Main().AlphaIsPremultiplied()
-              ? 1.f / std::max(kSmallAlpha, alpha)
-              : 1.f;
-      for (const float* const row : rows) {
-        pixel_data->push_back(BufferFormat<precision>::FromFloat(
-            std::max(0.f, std::min(1.f, alpha_multiplier * row[x]))));
-      }
-      if (has_alpha) {
-        pixel_data->push_back(
-            BufferFormat<precision>::FromFloat(255.f * alpha));
-      }
-    }
-  }
-}
+bool LoadJpegXlImage(const gchar *const filename, gint32 *const image_id) {
+  std::vector<uint8_t> icc_profile;
+  GimpColorProfile *profile = NULL;
```

Comment:
This is rather minor but could we use `nullptr`, and rename `loadlinear` to `load_linear`?

Reply:
No problem.  I'll make the change and update the commit.